dvc/repo/experiments/__init__.py

```diff
@@ -194,14 +194,47 @@ def _stash_exp(
                         if branch:
                             branch_name = ExpRefInfo.from_ref(branch).name
                         else:
-                            branch_name = ""
+                            branch_name = f"{resume_rev[:7]}"
                         if self.scm.is_dirty():
                             logger.info(
                                 "Modified checkpoint experiment based on "
                                 "'%s' will be created",
                                 branch_name,
                             )
                             branch = None
+                        elif (
+                            not branch
+                            or self.scm.get_ref(branch) != resume_rev
+                        ):
+                            msg = [
+                                (
+                                    "Nothing to do for unchanged checkpoint "
```

Comment:
"unchanged" here may be confusing since the last checkpoint can be continued from even if unchanged. I guess the point is that it's intermediate? Easiest solution: just remove "unchanged" in this line.